ppac-service-impl/src/main/kotlin/com/levi9/ppac/service/api/service/CompanyServiceImpl.kt

```diff
@@ -17,16 +20,68 @@ import java.util.*
 class CompanyServiceImpl(
     private val securityContext: SecurityContext<Int>,
     private val companyRepository: CompanyRepository,
-    private val codeRepository: CodeRepository
+    private val codeRepository: CodeRepository,
+    private val openingRepository: OpeningRepository,
+    private val companyCodeRepository: CompanyCodeRepository
 ) : CompanyService<Company> {
+
+    @Transactional
+    override fun addOpening(id: UUID, opening: Opening): Opening {
+
+        if (!codeRepository.isCompanyCode(securityContext.getAccessCode(), id)) {
```

Comment:
You can replace "if(!condition){throw ex}" with require(condition){throw ex}" . Apply this for all cases